R-packages/evalcast/R/plot_calibration.R

```diff
@@ -1,27 +1,53 @@
 #' Plot calibration curves
 #'
-#' @param scorecard Single score card.
+#' @param predictions_cards Either one predictions_card or several joined
+#'   together using bind_rows().
+#' @template geo_type-template 
 #' @param type One of "wedgeplot" or "traditional".
 #' @param grp_vars variables over which to compare calibration. These
 #'   determines the color of the lines and faceting depending on `type`
 #' @param avg_vars variables over which we average to determine the calibration.
 #' @param legend_position Legend position, the default being "bottom".
 #'
 #' @export
-plot_calibration <- function(scorecard,
+plot_calibration <- function(predictions_cards,
+                             geo_type,
                              type = c("wedgeplot", "traditional"),
                              grp_vars = c("forecaster", "forecast_date", "ahead"),
                              avg_vars = c("geo_value"),
                              legend_position = "bottom") {
   
   type <- match.arg(type)
   
-    
+  scorecard <- left_join(predictions_cards,
+                         get_covidcast_data(predictions_cards,
+                                            backfill_buffer,
+                                            geo_type),
+                         by = c("geo_value",
+                                "forecast_date",
+                                "ahead"))
+
   if (type == "wedgeplot") {
     grps <- grp_processing_for_facets(scorecard, grp_vars, 3, "wedgeplots")
-    ngrps <- length(grps)
-    calib <- compute_calibration(scorecard, grp_vars, avg_vars) %>%
-      setup_wedgeplot()
+    
+    if (length(grps) == 1) {
+      facet_cols <- c(grps[1])
+      facet_rows <- NULL
+    } else if (length(grps) == 2) {
+      facet_cols <- c(grps[2]) 
+      facet_rows <- c(grps[1])
+    } else {
+      facet_rows <- NULL
+      facet_cols <- NULL
+    }
+    
```

Comment:
I'd like to get rid of as much of this automatic faceting as possible. Instead put it in the function call the same as `plot_canonical()`.

Reply:
I tried doing that, but we would want grouping to be done differently for traditional and wedge plots in terms of facets and colors, so I don't know how we could have default arguments that would work for both.

Reply:
Three options then: 
1. Make wedge plots their own function that handles single forecasters only.
2. Change the wedge-plot colors to be only a single color and then allow multiple forecasters and faceting as with everything else.
3. Do both or maybe add a case to `plot_calibration()` to do either option 1 or 2 based on how many forecasters there are.

Reply:
Another idea, how about we leave it as is but have two wrapper functions for wedge plots and traditional plots?  Then we pass in ... to the plot_calibration, and have default values for the wedge plots and traditional plots for the facet_rows and facet_cols?   Then it's the same behavior in terms of using the function, just some internal refactoring to have default values for each.

Reply:
I did the idea I mentioned, lmk what you think.